src/async.js

```diff
@@ -162,15 +177,24 @@ Async.prototype._drainQueue = function(queue) {
         var receiver = queue.shift();
         var arg = queue.shift();
         fn.call(receiver, arg);
+        maxItemsToProcess--;
     }
+
+    return maxItemsToProcess;
 };
 
-Async.prototype._drainQueues = function () {
+Async.prototype._drainQueues = function (maxItemsToProcess) {
     ASSERT(this._isTickUsed);
-    this._drainQueue(this._normalQueue);
+    maxItemsToProcess = this._drainQueue(this._normalQueue, maxItemsToProcess);
     this._reset();
     this._haveDrainedQueues = true;
-    this._drainQueue(this._lateQueue);
+    this._drainQueue(this._lateQueue, maxItemsToProcess);
```

Comment:
I looked for latequeue usages and it seems its only used for debugging features like warnings/errors, and does not get used in mainline scenarios.